src/http/output/metadata/AllowAcceptMetadataWriter.ts

```diff
@@ -0,0 +1,99 @@
+import type { HttpResponse } from '../../../server/HttpResponse';
+import { MethodNotAllowedHttpError } from '../../../util/errors/MethodNotAllowedHttpError';
+import { NotFoundHttpError } from '../../../util/errors/NotFoundHttpError';
+import { UnsupportedMediaTypeHttpError } from '../../../util/errors/UnsupportedMediaTypeHttpError';
+import { addHeader } from '../../../util/HeaderUtil';
+import { isContainerPath } from '../../../util/PathUtil';
+import { LDP, PIM, RDF, SOLID_ERROR } from '../../../util/Vocabularies';
+import type { RepresentationMetadata } from '../../representation/RepresentationMetadata';
+import { MetadataWriter } from './MetadataWriter';
+
+/**
+ * Generates Allow, Accept-Patch, Accept-Post, and Accept-Put headers.
+ * The resulting values depend on the choses input methods and types.
+ * The input metadata also gets used to remove methods from that list
+ * if they are not valid in the given situation.
+ */
+export class AllowAcceptMetadataWriter extends MetadataWriter {
```

Comment:
I wonder if we want to start calling these classes
```suggestion
export class AllowAcceptHeaderWriter extends MetadataWriter {
```

Reply:
There might be instances where something else has to be done instead of writing a header? Although I can't immediately think of anything.

Reply:
Not for this class, we know it is a `AllowAcceptHeaderWriter`. (Would still keep `MetadataWriter` in general.)